src/ir_def/check_ir.ml

```diff
@@ -859,7 +858,7 @@ let rec check_exp env (exp:Ir.exp) : unit =
   if exp.note.Note.const
   then begin
     match exp.it with
-    | VarE id -> check_var "VarE" id
+    | VarE (Const, id) -> check_var "VarE" id
```

Comment:
So if the flag isn't Const, we now fall through? I guess that's ok

Reply:
Yes, we'd get "unexpected constant expression" below. Never happens.